### PR TITLE
Convert student accounts owned by teachers to teacher accounts.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Though I was initially leaning towards doing this in MySQL (for efficiency), after further reflection, I'd rather do it through ruby as ruby offers: cleaner documentation (source controlled), easier review and correctness verification, and not needing temporary DB tables in production.

Do speak up if you feel otherwise at this point.

### CODE REVIEW

File: bin/oneoff/data_fix_for_missing_emails [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  sleep(0.05)
+
+  user = User.find_by_email_or_hashed_email(email)
+  next if user.nil? || user.email.present?
+
+  begin
+    # As we know the email address to be associated with a teacher, we upgrade
+    # the account to a teacher account, populating the plaintext email address.
+    puts "UPDATING: #{user.id}..."
+    user.update(user_type: 'teacher', email: email)
```
Comment (aoby) [inline bin/oneoff/data_fix_for_missing_emails]: User::TYPE_TEACHER

File: bin/oneoff/data_fix_for_missing_emails [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#
+# Note that this script assumes a list of email addresses within the file
+# /tmp/teacher_emails.csv. Since this file contains PII, it will live only
+# temporarily (also not within source control) at that location.
+
+require 'csv'
+require_relative '../../dashboard/config/environment'
+
+CSV.foreach('/tmp/teacher_emails.csv') do |row|
+  email = row[0]
```
Comment (aoby) [inline bin/oneoff/data_fix_for_missing_emails]: Not that important but it looks like this isn't really a csv, but rather list of emails, one per line. You could do

``` ruby
File.open('/tmp/teacher_emails') do |line|
  email = line.strip
...
```

Comment (ashercodeorg) [inline bin/oneoff/data_fix_for_missing_emails]: Done.

File: bin/oneoff/data_fix_for_missing_emails (hunk 1 above)
Comment (ashercodeorg) [inline bin/oneoff/data_fix_for_missing_emails]: Done.

**Comment by aoby** [comment]
LGTM

**Pull request merged**